pypdf/_writer.py

```diff
@@ -833,6 +833,23 @@ def _update_field_annotation(
                 ),
             ).get_object(),
         )
+
+        if "/Font" not in dr or not isinstance(dr["/Font"], DictionaryObject):
+            dr[NameObject("/Font")] = DictionaryObject()
+
+        font_dict = dr["/Font"]
+
+        # Check if the specific font (e.g., Helvetica) is in /Font
+        font_name = NameObject("/Helvetica")
+        if font_name not in font_dict:
+            font_entry = DictionaryObject({
+                NameObject("/Type"): NameObject("/Font"),
+                NameObject("/Subtype"): NameObject("/Type1"),
+                NameObject("/BaseFont"): NameObject("/Helvetica"),
```

Comment:
Would there be any side effect of re-using `font_name` here?

Reply:
Good catch, I will fix this.